src/crimson/os/seastore/transaction.h

```diff
@@ -605,6 +580,31 @@ class Transaction {
     }
   }
 
+  auto lookup_read_set(CachedExtentRef ref) const {
```

Comment:
Related to https://github.com/ceph/ceph/pull/62071 : we could change the type of parameter to `const CachedExtent &extent` that is required by `CachedExtent::is_viewable_by_trans()`

```cpp
  /**
   * is_viewable_by_trans
   *
   * Check if this extent is still viewable by transaction t.
   *
   * Precondition: *this was previously visible to t, which indicates
   * this extent is either in the read set of t or created(pending) by t.
   */
  std::pair<bool, viewable_state_t>
  CachedExtent::is_viewable_by_trans(Transaction &t) const {
    // ...
    assert(t.is_in_read_set(*this));
    // ...
  }
```

Reply:
Seems fine to adjust the type whenever necessary.